docs/docs/conceptual/fast-refresh.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Fast Refresh
+---
+
+Fast Refresh is a feature that allows you to get direct feedback for changes in your React components. Fast Refresh is enabled by default in Gatsby's development server and should show your changes within a second, without losing component state.
+
+## How It Works
+
+- If you edit a file that only exports React component(s), Fast Refresh will update the code only for that file, and re-render your component. You can edit anything in that file, including styles, rendering logic, event handlers, or effects
+- If you edit a file with exports that aren't React components, Fast Refresh will re-run both that file, and the other files importing it. So if both `Button.js` and `Modal.js` import `theme.js`, editing `theme.js` will update both components
+- Finally, if you edit a file that's imported by files outside of the React tree, Fast Refresh will fall back to doing a full reload. You might have a file which renders a React component but also exports a value that is imported by a non-React component. For example, maybe your component also exports a constant, and a non-React utility file imports it. In that case, consider migrating the constant to a separate file and importing it into both files. This will re-enable Fast Refresh to work. Other cases can usually be solved in a similar way
+
+## Error Resilience
+
+Fast Refresh comes with a good resilience against errors and the custom overlay in Gatsby will provide you with information on how to resolve those. Whenever possible a code frame showing the exact error location, including the original line is given. The overlay will indicate whether it's a compile error, [GraphQL](/docs/glossary#graphql) error, or runtime error. Once you fixed the error the overlay will automatically disappear.
```

Comment:
```suggestion
Fast Refresh is resilient against errors and the custom overlay in Gatsby will provide you with information on how to resolve those. Whenever possible, a code frame showing the exact error location, including the original line, is given. The overlay will indicate whether it's a compile error, [GraphQL](/docs/glossary#graphql) error, or runtime error. Once you fix the error the overlay will automatically disappear.
```
I'm not sure resilience is the word here. What are you trying to say? That they're compatible and facilitate errors well? Or that fast refresh ignores errors?

Reply:
I guess what I was trying to say is:

For most (but hopefully all) errors our Fast Refresh integration catches those and shows you the custom overlay instead of silently failing/showing a blank page. Furthermore, after fixing those things it recovers itself from that state and goes away.

Reply:
Ah ok. So resiliant makes sense but I think it's a bit hard to understand unless you already know what's happening. So perhaps keep the section title but use that longer explanation? "The Fast Refresh integration is good at catching errors instead of silently failing. Gatsby takes this opportunity to present you with a custom overlay that further explains the error and how to resolve it. Once it's resolved Fast refresh will recover from that state."